tests/UnifiedSpecTests/ManagesFailPointsTrait.php

```diff
@@ -31,8 +32,14 @@ public function configureFailPoint(stdClass $failPoint, Server $server): void
     public function disableFailPoints(): void
     {
         foreach ($this->failPointsAndServers as [$failPoint, $server]) {
-            $operation = new DatabaseCommand('admin', ['configureFailPoint' => $failPoint, 'mode' => 'off']);
-            $operation->execute($server);
+            try {
+                $operation = new DatabaseCommand('admin', ['configureFailPoint' => $failPoint, 'mode' => 'off']);
+                $operation->execute($server);
+            } catch (ConnectionException $e) {
+                // Retry once in case the connection was dropped by the last operation
```

Comment:
I'm still at a loss for why this wasn't an issue with the original FailPointObserver. There, the Server object was instantiated from the CommandStartedEvent; however, it's just an integer `server_id` and Manager reference. That's not much different from the object we're capturing here from `configureFailPoint()`. In both cases, the command to disable a fail point is still being executed via `phongo_execute_command()` with the same underlying `mongoc_client_t` and `server_id`.

The only difference I can identify is _when_ we're disabling fail points. Previously, FailPointObserver disabled fail points during our equivalent of a `tearDown()` method. Now, we're doing so immediately after executing operations and before evaluating `expectEvents` (note: this is what the spec itself suggests). It's possible the previous deferral allowed some opportunity to reestablish a dropped connection.

This may be related to [CDRIVER-4532](https://jira.mongodb.org/browse/CDRIVER-4532), which highlights an outstanding libmongoc deficiency in the code paths utilized by PHPC.